Library/OcDevicePathLib/OcDevicePathLib.c

```diff
@@ -1090,7 +1093,7 @@ OcFileDevicePathFullNameSize (
   IN CONST EFI_DEVICE_PATH_PROTOCOL  *DevicePath
   )
 {
-  return (OcFileDevicePathFullNameLen (DevicePath) + 1) * sizeof (CHAR16);
```

Comment:
It looks like the contract of this method `@retval 0  DevicePath does not start with a File Path node or contains non-terminating nodes that are not File Path nodes` was not being met before? (The return value is the same as before, with this change, when there is a path, but would have been 1 with no path.)

Reply:
I don't understand? We return 0 if we encounter a non-file node except for the DP terminator.

Reply:
_This_ method `OcFileDevicePathFullNameSize` (if you expand up, I could not add this comment on the right line, really, as GitHub isn't showing it as part of the diff lines).

So previously (and now!) `OcFileDevicePathFullNameLen` returns 0 for non-file, and therefore _this_ method `OcFileDevicePathFullNameSize` (as it was) returned 2 (`(0 + 1) * sizeof (CHAR16)`) at this point.

Now, with this change, both will return 0 (`0 * sizeof (CHAR16)` from here).

I believe it is correct now and was wrong before?

Reply:
(Just to be clear, I'm basing this on the @retval comment if this method, which I already quoted - which *says* this method will return 0 if not a file, but it would actually have returned 2 before. I think?!)

Reply:
Ugh, yes, GitHub diff messed with me. Thank you :)